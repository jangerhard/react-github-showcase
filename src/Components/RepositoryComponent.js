import PropTypes from 'prop-types';
import React from 'react';
import CommitComponent from "./CommitComponent";

const RepositoryComponent = ({ name, commits, url }) => (
    <div>
        <a href={url}> {name} </a>
        {!commits.length && <NoCommitMessage/>}
        {commits.map((commit) => {
            return <CommitComponent
                key={commit.abbreviatedOid}
                abbreviatedOid={commit.abbreviatedOid}
                message={commit.message}
                committedDate={commit.committedDate} />;
        })}
    </div>
);

RepositoryComponent.propTypes = {
    name: PropTypes.string.isRequired,
    commits: PropTypes.array.isRequired,
    url: PropTypes.string.isRequired
};

const NoCommitMessage = () => (
    <div>
        No commits yet.
    </div>
);

export default RepositoryComponent;
