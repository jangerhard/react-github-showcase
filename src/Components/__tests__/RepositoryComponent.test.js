import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import RepositoryComponent from "../RepositoryComponent";

const name = 'Test Testington';
const commits = [
    {
        abbreviatedOid: 'AAAAAA',
        message: 'This is a commit message',
        committedDate: '2018-05-04T19:59:54Z',
    }
];
const url = 'www.test.com';

test('renders repository correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <RepositoryComponent
            commits={commits}
            name={name}
            url={url}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});

test('renders repository with no commits correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <RepositoryComponent
            commits={[]}
            name={name}
            url={url}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});