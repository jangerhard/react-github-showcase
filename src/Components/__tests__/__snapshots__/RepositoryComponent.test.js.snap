// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders repository correctly 1`] = `
<div>
  <a
    href="www.test.com"
  >
     
    Test Testington
     
  </a>
  <CommitComponent
    abbreviatedOid="AAAAAA"
    committedDate="2018-05-04T19:59:54Z"
    message="This is a commit message"
  />
</div>
`;

exports[`renders repository with no commits correctly 1`] = `
<div>
  <a
    href="www.test.com"
  >
     
    Test Testington
     
  </a>
  <div
    style={
      Object {
        "paddingLeft": "1em",
      }
    }
  >
    No commits yet.
  </div>
</div>
`;
